### PR TITLE
RGRIDT-954: Performance Measure API

### PR DESCRIPTION
**Pull request by tmbp95**
This PR is for RGRIDT-954: Performance Measure API

### What was happening
* We need a way to measure our performance in runtime

### What was done
* Created a set of new classes responsible for taking all the metrics we need to measure the performance of our code:
  * PerformanceAPI  - responsible for all the methods that we need to invoke. Allows setting marks on the code, getting measures from two marks, and also getting all the measures that were created in runtime in order to get a final object containing all those metrics.
  * Two new PerformanceModes  - PerformanceDevMode and PerformanceProdMod.
  * AbstractPerformance and IPerformance - the skeleton of the PerformanceDevMode and PerformanceProdMode

### Screenshots
image

### Checklist
* [ ] tested locally - NA
* [x] documented the code
* [x] clean all warnings and errors of eslint
* [ ] requires changes in OutSystems (if so, provide a module with changes) - NA
* [ ] requires new sample page in OutSystems (if so, provide a module with changes) - NA

### CODE REVIEW

**Review by glundgren93 — APPROVED** [review]

File: code/src/GridAPI/ConditionalFormat.ts [hunk 1]
```diff
@@ -13,11 +13,22 @@ namespace GridAPI.ConditionalFormat {
...
                 gridObj.features.conditionalFormat.addRules(binding, rules);
+
+                PerformanceAPI.SetMark(
+                    'ConditionalFormat.AddConditionalFormat-end'
+                );
+                PerformanceAPI.GetMeasure(
+                    '@datagrid-ConditionalFormat.AddConditionalFormat',
+                    'ConditionalFormat.AddConditionalFormat',
+                    'ConditionalFormat.AddConditionalFormat-end'
+                );
```
Comment (rugoncalves) [inline code/src/GridAPI/ConditionalFormat.ts:31]: What are we measuring here?

File: code/src/GridAPI/Pagination.ts [hunk 2]
```diff
@@ -173,6 +258,13 @@ namespace GridAPI.Pagination {
                     OSFramework.Enum.PageLabel.PageSize,
                     phID
                 );
+
+                PerformanceAPI.SetMark('Pagination.RegisterPageSizeLabel-end');
+                PerformanceAPI.GetMeasure(
+                    '@datagrid-Pagination.RegisterPageSizeLabel',
+                    'Pagination.RegisterPageSizeLabel',
+                    'Pagination.RegisterPageSizeLabel-end'
+                );
```
[the remark this answers is not in the document]
Comment (glundgren93) [reply code/src/GridAPI/Pagination.ts:263]: fixed

File: code/src/GridAPI/Pagination.ts [hunk 3]
```diff
@@ -39,6 +50,13 @@ namespace GridAPI.Pagination {
                     phID,
                     buttonQuantity
                 );
+
+                PerformanceAPI.SetMark('Pagination.CreatePageButtons-end');
+                PerformanceAPI.GetMeasure(
+                    '@datagrid-Pagination.CreatePageButtons',
+                    'Pagination.CreatePageButtons',
+                    'Pagination.CreatePageButtons-end'
+                );
```
[the remark this answers is not in the document]
Comment (glundgren93) [reply code/src/GridAPI/Pagination.ts:58]: fixed

File: code/src/GridAPI/ConditionalFormat.ts [hunk 4]
```diff
@@ -33,11 +44,22 @@ namespace GridAPI.ConditionalFormat {
...
                 gridObj.features.conditionalFormat.removeRules(binding);
+
+                PerformanceAPI.SetMark(
+                    'ConditionalFormat.RemoveConditionalFormat-end'
+                );
+                PerformanceAPI.GetMeasure(
+                    '@datagrid-ConditionalFormat.RemoveConditionalFormat',
+                    'ConditionalFormat.RemoveConditionalFormat',
+                    'ConditionalFormat.RemoveConditionalFormat-end'
+                );
```
[the remark this answers is not in the document]
Comment (glundgren93) [reply code/src/GridAPI/ConditionalFormat.ts:64]: fixed

File: code/src/GridAPI/ConditionalFormat.ts (hunk 1 above)
Comment (glundgren93) [reply code/src/GridAPI/ConditionalFormat.ts:32]: fixed

**Review by glundgren93 — APPROVED** [review]

**Pull request merged**